Source/Scene/Cesium3DTileBatchTable.js

```diff
@@ -0,0 +1,1519 @@
+            setColor +
+            applyHighlight +
+            '} \n';
+
+        return source;
+    }
+
+    Cesium3DTileBatchTable.prototype.getFragmentShaderCallback = function(handleTranslucent, diffuseUniformName) {
+        if (this.featuresLength === 0) {
+            return;
+        }
+        return function(source) {
+            source = modifyDiffuse(source, diffuseUniformName);
+            if (ContextLimits.maximumVertexTextureImageUnits > 0) {
+                // When VTF is supported, per-feature show/hide already happened in the fragment shader
+                source +=
+                    'varying vec4 tile_featureColor; \n' +
+                    'void main() \n' +
+                    '{ \n' +
+                    '    tile_color(tile_featureColor); \n' +
+                    '}';
+            } else {
+                source +=
+                    'uniform sampler2D tile_batchTexture; \n' +
+                    'uniform bool tile_translucentCommand; \n' +
+                    'varying vec2 tile_featureSt; \n' +
+                    'void main() \n' +
+                    '{ \n' +
+                    '    vec4 featureProperties = texture2D(tile_batchTexture, tile_featureSt); \n' +
+                    '    if (featureProperties.a == 0.0) { \n' + // show: alpha == 0 - false, non-zeo - true
+                    '        discard; \n' +
+                    '    } \n';
+
+                if (handleTranslucent) {
+                    source +=
+                        '    bool isStyleTranslucent = (featureProperties.a != 1.0); \n' +
+                        '    if (czm_pass == czm_passTranslucent) \n' +
+                        '    { \n' +
+                        '        if (!isStyleTranslucent && !tile_translucentCommand) \n' + // Do not render opaque features in the translucent pass
+                        '        { \n' +
+                        '            discard; \n' +
+                        '        } \n' +
+                        '    } \n' +
+                        '    else \n' +
+                        '    { \n' +
+                        '        if (isStyleTranslucent) \n' + // Do not render translucent features in the opaque pass
+                        '        { \n' +
+                        '            discard; \n' +
+                        '        } \n' +
+                        '    } \n';
+                }
+
+                source +=
+                    '    tile_color(featureProperties); \n' +
+                    '} \n';
+            }
+            return source;
+        };
+    };
+
+    function getColorBlend(batchTable) {
+        var tileset = batchTable._content._tileset;
+        var colorBlendMode = tileset.colorBlendMode;
+        var colorBlendAmount = tileset.colorBlendAmount;
+        if (colorBlendMode === Cesium3DTileColorBlendMode.HIGHLIGHT) {
+            return 0.0;
+        } else if (colorBlendMode === Cesium3DTileColorBlendMode.REPLACE) {
```

Comment:
No need to use else, since they all return, also If none of these match, should it be a runtime error?